sdk/java-sdk-protobuf/src/main/java/kalix/javasdk/JsonSupport.java

```diff
@@ -214,14 +230,10 @@ public static <T, C extends Collection<T>> C decodeJsonCollection(Class<T> value
         ByteString decodedBytes = ByteStringEncoding.decodePrimitiveBytes(any.getValue());
         var typeRef = objectMapper.getTypeFactory().constructCollectionType(collectionType, valueClass);
         return objectMapper.readValue(decodedBytes.toByteArray(), typeRef);
+      } catch (JsonProcessingException e) {
+        throw jsonProcessingException(valueClass, any, e);
       } catch (IOException e) {
-        throw new IllegalArgumentException(
-          "JSON with type url ["
-            + any.getTypeUrl()
-            + "] could not be decoded into a ["
-            + valueClass.getName()
-            + "]",
-          e);
+        throw genericDecodeException(valueClass, any, e);
```

Comment:
In which case would we get a `IOException` that is not a `JsonProcessingException`? 
Corrupted payload?

Reply:
Ok, maybe when reading from an external topic that is emitting bad data. In such a case, there is nothing the user can do. Migration won't work anyway.